x/txfees/module.go

```diff
@@ -162,3 +168,45 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
 
 // ConsensusVersion implements AppModule/ConsensusVersion.
 func (AppModule) ConsensusVersion() uint64 { return 1 }
+
+// On start, we unmarshal the consensus params once and cache them.
+// Then, on every block, we check if the current consensus param bytes have changed in comparison to the cached value.
+// If they have, we unmarshal the current consensus params, update the target gas, and cache the value.
+// This is done to improve performance by not having to fetch and unmarshal the consensus params on every block.
+func (am AppModule) CheckAndSetTargetGas(ctx sdk.Context) {
+	// Check if the block gas limit has changed.
+	// If it has, update the target gas for eip1559.
+	consParamsBytes := am.keeper.GetParamsNoUnmarshal(ctx)
+
+	// If cachedConsParamBytes is nil, set equal to consParamsBytes and set the target gas.
+	if cachedConsParamBytes == nil {
+		cachedConsParamBytes = consParamsBytes
+		newConsensusParams, err := am.keeper.UnmarshalParamBytes(ctx, consParamsBytes)
+		if err != nil {
+			panic(err)
+		}
+
+		// Check if newConsensusParams.Block is nil to prevent panic
+		if newConsensusParams.Block == nil || newConsensusParams.Block.MaxGas == 0 {
+			return
+		}
+
+		newBlockMaxGas := newConsensusParams.Block.MaxGas
+		mempool1559.TargetGas = newBlockMaxGas
```

Comment:
This isn't right, it should be max gas times .625

Reply:
Done here https://github.com/osmosis-labs/osmosis/pull/7270/commits/d148048eb9fbc70814d843236240e73f30787ebd